packages/pages/src/common/src/parsers/puckConfigParser.ts

```diff
@@ -0,0 +1,52 @@
+import fs from "node:fs";
+import SourceFileParser, { createTsMorphProject } from "./sourceFileParser.js";
+import { newConfig } from "../../../scaffold/template/sampleTemplates.js";
+import { SyntaxKind } from "ts-morph";
+
+/**
+ * ConfigParser is a class for parsing the puck config file.
+ */
+export default class ConfigParser {
+  /**
+   * Adds variables to the puckConfig file and adds the new config to
+   * the exported map.
+   * @param templateName as recieved by user input
```

Comment:
I don't think we need all 3 of these. We should just take the user's input for templateName, cleanse it, and pass/use that one thing for everything: the filename, and template name in the TemplateConfig. Technically the name isn't even necessary in the TemplateConfig because the filename will just be used, but we can add it for completeness.